crates/ty_server/src/server.rs

```diff
@@ -170,13 +172,80 @@ impl Server {
 
     fn event_loop(
         connection: &Connection,
-        _client_capabilities: &ClientCapabilities,
+        client_capabilities: &ClientCapabilities,
         mut session: Session,
         worker_threads: NonZeroUsize,
     ) -> crate::Result<()> {
         let mut scheduler =
             schedule::Scheduler::new(&mut session, worker_threads, connection.make_sender());
 
+        let fs_watcher = client_capabilities
+            .workspace
+            .as_ref()
+            .and_then(|workspace| workspace.did_change_watched_files?.dynamic_registration)
+            .unwrap_or_default();
+
+        if fs_watcher {
+            let registration = lsp_types::Registration {
+                id: "workspace/didChangeWatchedFiles".to_owned(),
+                method: "workspace/didChangeWatchedFiles".to_owned(),
+                register_options: Some(
+                    serde_json::to_value(DidChangeWatchedFilesRegistrationOptions {
+                        watchers: vec![
+                            FileSystemWatcher {
+                                glob_pattern: lsp_types::GlobPattern::String("**/ty.toml".into()),
+                                kind: None,
+                            },
+                            FileSystemWatcher {
+                                glob_pattern: lsp_types::GlobPattern::String(
+                                    "**/.gitignore".into(),
+                                ),
+                                kind: None,
+                            },
+                            FileSystemWatcher {
+                                glob_pattern: lsp_types::GlobPattern::String("**/.ignore".into()),
+                                kind: None,
+                            },
```

Comment:
If we want to be (more) exhaustive here, we would probably also have to add `.git/info/exclude`?

Reply:
Maybe. It's not something that we currently handle in `apply_changes`. (but maybe should?)

Reply:
I'll ignore this for now. There's another issue around better handling for git ignored files